src/wasm/wasm.cpp

```diff
@@ -210,7 +210,7 @@ void Block::finalize() {
 
 void If::finalize(WasmType type_) {
   type = type_;
-  if (type == none && (condition->type == unreachable || (ifFalse && ifTrue->type && ifFalse->type == unreachable))) {
+  if (type == none && (condition->type == unreachable || (ifFalse && ifTrue->type == unreachable && ifFalse->type == unreachable))) {
```

Comment:
No, because that's the same as having an empty else arm. So if the condition is not taken, we do reach the outside.